osaca/osaca.py

```diff
@@ -305,12 +301,8 @@ def write_csv(self, csv):
         csv : str
             CSV data as string
         """
-        #curr_dir = '/'.join(os.path.realpath(__file__).split('/')[:-1])
-        try:
-            f = open(self.osaca_dir+'data/'+self.arch.lower()+'_data.csv', 'w')
-        except IOError:
-            print('IOError: file \'{}\' not found in $HOME/.osaca/data'.format(self.arch.lower()
-                  + '_data.csv'), file=self.file_output)
+        # curr_dir = '/'.join(os.path.realpath(__file__).split('/')[:-1])
+        f = open(self.osaca_dir + 'data/' + self.arch.lower() + '_data.csv', 'w')
```

Comment:
@cod3monk Why did you remove the error message here? I think it is helpful in case of something went wrong...

Reply:
Because the catch block only prints an error message and then does nothing to resolve the issue. The code execution will therefore be continued and will fail (with another exception) on `f.write(csv)`, which is misleading because it does not point at the actual problem. By not handling the exception at all, execution will be stopped and the user is presented with a very similar error message to yours (plus a stack trace for easy debugging, if needed).
If a `sys.exit(1)` would have been added to the catch block, it would have been fine with me.